modules/macos_run_puppet/files/run-puppet.sh

```diff
@@ -0,0 +1,196 @@
+
+# Require override settings file
+SETTINGS_FILE="/opt/puppet_environments/ronin_settings"
+
+if [ ! -f "$SETTINGS_FILE" ]; then
+    echo "Missing required settings file: $SETTINGS_FILE"
+    exit 1
+fi
+
+# shellcheck source=/opt/puppet_environments/ronin_settings
+echo "Loading settings from $SETTINGS_FILE..."
+source "$SETTINGS_FILE"
+
+# Expect these to be set in the sourced file
+# PUPPET_REPO, PUPPET_BRANCH, PUPPET_ROLE_FILE, PUPPET_BIN, FACTER_BIN
+
+export GIT_REPO_URL="$PUPPET_REPO"
+export GIT_BRANCH="$PUPPET_BRANCH"
+
+GIT_USERNAME=$(extract_username_from_url "$GIT_REPO_URL")
+LOCAL_PUPPET_REPO="/opt/puppet_environments/${GIT_USERNAME}/ronin_puppet"
+
+echo "Using Puppet Repo: $GIT_REPO_URL"
+echo "Using Branch: $GIT_BRANCH"
+
+fail() {
+    echo "${@}"
+    exit 1
+}
+
+# Ensure Puppet Role is Set
+if [ -f "$PUPPET_ROLE_FILE" ]; then
+    ROLE=$(<"$PUPPET_ROLE_FILE")
+else
+    fail "Failed to find Puppet role file $PUPPET_ROLE_FILE"
+fi
+
+# Ensure Puppet & Facter are Installed
+if [ ! -x "$PUPPET_BIN" ]; then
+    fail "Puppet is missing or not executable."
+fi
+
+if [ ! -x "$FACTER_BIN" ]; then
+    fail "Facter is missing or not executable."
+fi
+
+# Clone or update Puppet repository
+if [ -d "$LOCAL_PUPPET_REPO/.git" ]; then
+    echo "Checking existing Puppet repository..."
+    cd "$LOCAL_PUPPET_REPO" || fail "Failed to change directory to Puppet repository"
+
+    CURRENT_REMOTE_URL=$(git remote get-url origin 2>/dev/null)
+
+    if [ "$CURRENT_REMOTE_URL" != "$GIT_REPO_URL" ]; then
+        echo "Repository URL has changed. Removing old repository..."
+        cd ..
+        rm -rf "$LOCAL_PUPPET_REPO"
+        echo "Cloning new Puppet repository..."
+        git clone --branch "$GIT_BRANCH" "$GIT_REPO_URL" "$LOCAL_PUPPET_REPO" || fail "Failed to clone Puppet repository"
+    else
+        echo "Checking for updates in Puppet repository..."
+        git fetch origin "$GIT_BRANCH" || fail "Failed to fetch latest changes"
+
+        LOCAL_COMMIT=$(git rev-parse HEAD)
+        REMOTE_COMMIT=$(git rev-parse "origin/$GIT_BRANCH")
+
+        if [ "$LOCAL_COMMIT" != "$REMOTE_COMMIT" ]; then
+            echo "Updates found. Pulling latest changes..."
+            git reset --hard "origin/$GIT_BRANCH" || fail "Failed to reset to latest commit"
+        else
+            echo "Already up-to-date. No changes needed."
+        fi
+    fi
+else
+    echo "Cloning fresh Puppet repository..."
+    mkdir -p "$(dirname "$LOCAL_PUPPET_REPO")"
+    git clone --branch "$GIT_BRANCH" "$GIT_REPO_URL" "$LOCAL_PUPPET_REPO" || fail "Failed to clone Puppet repository"
+fi
+
+# Ensure Puppet Repository Exists
+get_puppet_repo() {
+    if [ ! -d "$LOCAL_PUPPET_REPO" ]; then
+        fail "Local Puppet repository not found at $LOCAL_PUPPET_REPO"
+    fi
+    cd "$LOCAL_PUPPET_REPO" || fail "Failed to change directory to $LOCAL_PUPPET_REPO"
+
+    # Inject Hiera Secrets
+    mkdir -p ./data/secrets
+    cp /var/root/vault.yaml ./data/secrets/vault.yaml
```

Comment:
let's do something like `chmod 0600 "${WORKING_DIR}/data/secrets/vault.yaml"` (from the linux run-puppet.sh script) after the copy to just be sure the perms are locked down on these secrets (we shouldn't rely on sane umask values for root being set).